### PR TITLE
[reciever/awsxrayreceiver] removed deprecated net.Err.Temporary()

### PR DESCRIPTION
**Pull request by pavankrish123**
**Description:** Removed deprecated net.Err.Temporary() reference from the code as mentioned in #9782.  As per https://github.com/golang/go/issues/45729 TimeOut() correctly captures subset of Temporary() errors that could be retried. The rest of Temporary() errors should not be retried anyways (like syscall errors, out of file descriptors) as described [here](https://github.com/golang/go/issues/45729#issue-866319435) . 

**Link to tracking Issue:** #9806 

**Testing:** Unit tests

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 ### 🛑 Breaking changes 🛑
 
 - `jmxreceiver`: Remove properties & groovyscript parameters from JMX Receiver. Add ResourceAttributes & LogLevel parameter to supply some of the removed functionality with reduced attack surface (#9685)
-
+- `awsxrayreceiver`: Removed deprecated `net.Error.Temporary()` reference from code.
```
Comment (dmitryax) [inline CHANGELOG.md:8]: Is this really a breaking change? I think we can use the `Skip Changelog` label in this PR

Comment (pavankrish123) [reply CHANGELOG.md:8]: Not really - it is completely hidden , I am happy to remove Changelog

Comment (pavankrish123) [reply CHANGELOG.md]: @dmitryax removed changelog

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,7 +5,6 @@
 ### 🛑 Breaking changes 🛑
 
 - `jmxreceiver`: Remove properties & groovyscript parameters from JMX Receiver. Add ResourceAttributes & LogLevel parameter to supply some of the removed functionality with reduced attack surface (#9685)
```
Comment (dmitryax) [inline CHANGELOG.md:7]: Bringing back the removed line

```suggestion
- `jmxreceiver`: Remove properties & groovyscript parameters from JMX Receiver. Add ResourceAttributes & LogLevel parameter to supply some of the removed functionality with reduced attack surface (#9685)

```

File: receiver/awsxrayreceiver/internal/udppoller/poller_test.go [hunk 3]
```diff
@@ -381,19 +381,19 @@ func TestSocketGenericReadError(t *testing.T) {
...
 
 func (m *mockNetError) Timeout() bool {
-	return false
+	return m.timeout
 }
 
 func (m *mockNetError) Temporary() bool {
-	return m.temporary
+	return false
 }
```
Comment (dmitryax) [inline receiver/awsxrayreceiver/internal/udppoller/poller_test.go:397]: Can this be removed?

Comment (pavankrish123) [reply receiver/awsxrayreceiver/internal/udppoller/poller_test.go:397]: if you are referring to the removing Temporary() function completely - mockNetError is implementing net.Error interface so it is required to implement the function even if it is not doing anything. Let me know if that is your intention.

Comment (dmitryax) [reply receiver/awsxrayreceiver/internal/udppoller/poller_test.go:397]: Ok. Sounds good

**Review by dmitryax — APPROVED** [review]

**Comment by codeboten** [comment]
Pinging codeowner for approval @willarmiros

**Review by willarmiros — APPROVED** [review]
lgtm!

**Review by codeboten — APPROVED** [review]

**Pull request merged**